File: algorithms/sort/BubbleSort.ts
export default function BubbleSort(array: number[]): number[] {
    const length = array.length

    for (let i = 0; i < length; i++) {
        for (let j = 0; j < length - 1 - i; j++) {
            if (array[j] > array[j + 1]) {
                let temp = array[j + 1]
                array[j + 1] = array[j]
                array[j] = temp
            }
        }
    }

    return array
}

console.log(BubbleSort([12, 1, 21, 5 + 6, 78, 89, 54, 56, 5, 3]))

File: algorithms/sort/BubbleSort.go
package sort

import "fmt"

func BubbleSort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		for j := 0; j < len(arr)-1-i; j++ {
			if arr[j] > arr[j+1] {
				tmp := arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = tmp
			}
		}
	}

	return arr
}

func RunBubbleSort() {
	array := []int{12, 5, 6, 56, 89, 821, 6, 56, 23, 45}

	fmt.Println(BubbleSort(array))
}
